### PR TITLE
Add display name and email management to account settings page

### PR DESCRIPTION
**Pull request by gentamura**
## Summary
This pull request introduces several significant changes to the account settings functionality, including the addition of a new form for updating the display name, enhancements to account information retrieval, and a script for synchronizing email addresses.

## Related Issue
N/A

## Changes

 The most important changes are summarized below:

### New Feature: Display Name Update

* [`app/(main)/settings/account/account-display-name-form.tsx`](diffhunk://#diff-85b14266a6fbc3146dd3b49c54bdcc876ac5a6c15c29cd7da4eb5f3ba6eebf2eR1-R118): Added a new form component `AccountDisplayNameForm` for updating the user's display name with validation and state management.
* [`app/(main)/settings/account/actions.ts`](diffhunk://#diff-94aaedffd97c760d2c37601f22744a7481abd96bf1d9095330802b9c286d56caR109-R152): Implemented `updateDisplayName` function to handle the display name update logic and database interaction.

### Enhancements to Account Information Retrieval

* [`app/(main)/settings/account/actions.ts`](diffhunk://#diff-94aaedffd97c760d2c37601f22744a7481abd96bf1d9095330802b9c286d56caR109-R152): Added `getAccountInfo` function to retrieve account information from the database.
* [`app/(main)/settings/account/page.tsx`](diffhunk://#diff-7af88939f7f40e69dc5aa0b7532e8c669e3f8dedf5c06c8f410427d4d6aa56e8R3-R17): Updated the account settings page to use `getAccountInfo` for fetching display name and email, and integrated `AccountDisplayNameForm` for display name updates. [[1]](diffhunk://#diff-7af88939f7f40e69dc5aa0b7532e8c669e3f8dedf5c06c8f410427d4d6aa56e8R3-R17) [[2]](diffhunk://#diff-7af88939f7f40e69dc5aa0b7532e8c669e3f8dedf5c06c8f410427d4d6aa56e8L24-R37)

### Database Synchronization Script

* [`scripts/20241129-db-patch-add-name-and-email-to-users.ts`](diffhunk://#diff-ff1d9dea2b499a4360142922793464a13d035c8fe581a13045fb77fc009fce6dR1-R75): Added a script to update email addresses in the users table by synchronizing with Supabase Auth, ensuring the email addresses are up-to-date.
* Added email field update in the `initializeAccount` function to ensure email is set during account initialization. (`services/accounts/actions/initialize-account.ts`)

## Testing
- Successfully tested form validation for display name updates
- Verified email synchronization script with test data
- Confirmed proper error handling in account information retrieval
- Tested account settings page with various user scenarios
- Verified database updates are properly reflected in the UI

## Other Information
- Prior implementation has been completed and verified in PR #148
- Production database updates will be completed through script execution
- The script can be safely executed either before or after merging this PR, with no impact on the deployment sequence

### CODE REVIEW

File: app/(main)/settings/account/actions.ts [hunk 1]
```diff
@@ -103,3 +106,47 @@ export async function disconnectGoogleIdentity() {
...
+
+		const userDbIdSubquery = db
+			.select({ userDbId: supabaseUserMappings.userDbId })
+			.from(supabaseUserMappings)
+			.where(eq(supabaseUserMappings.supabaseUserId, supabaseUser.id));
+
+		await db
+			.update(users)
+			.set({ displayName })
+			.where(eq(users.dbId, userDbIdSubquery));
```
Comment (gentamura) [inline app/(main)/settings/account/actions.ts:148]: The reason I used this subquery is that drizzle's `eq` does not support fields containing `null`.
By the way, if `users.email` is `not null`, I could just write `eq(users.email, supabaseUser.email)`.

**Comment by gentamura** [comment]
@satococoa review, please 🙏

**Comment by gentamura** [comment]
@shige After review, please run the script in the service role as follows.
https://github.com/giselles-ai/giselle/pull/124#issuecomment-2510562449

The target script is `bun run scripts/20241129-db-patch-add-name-and-email-to-users.ts`. :pray:

File: scripts/20241129-db-patch-add-name-and-email-to-users.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	// Filter out users without email
+	const validUsers = users.filter((user) => user.email);
+
+	if (validUsers.length === 0) {
+		console.log("No users to update");
+		return;
+	}
+
+	// Perform bulk update
```
Comment (satococoa) [inline scripts/20241129-db-patch-add-name-and-email-to-users.ts]: The comment `// Perform bulk update` might be a bit misleading. Bulk update usually implies updating everything with a single SQL command, but it looks like the updates are being done one at a time.

```suggestion
	// Perform update for all valid users
```

File: app/(main)/settings/account/actions.ts (hunk 1 above)
Comment (satococoa) [reply app/(main)/settings/account/actions.ts:148]: This upcoming feature in drizzle might be useful for our case:
https://github.com/drizzle-team/drizzle-orm/pull/2963

However, our current implementation using subqueries is sufficient for our needs 👍

**Review by satococoa — APPROVED** [review]
Thank you! 👍

**Comment by gentamura** [comment]
@shige @toyamarinyon review, please :pray:

**Review by toyamarinyon — APPROVED** [review]

**Review by shige — APPROVED** [review]
LGTM 👍

**Comment by shige** [comment]
https://github.com/giselles-ai/giselle/pull/144#issuecomment-2513613662

> The target script is `bun run scripts/20241129-db-patch-add-name-and-email-to-users.ts`. 🙏

All done! ✅

**Pull request merged**